### PR TITLE
refactor: replace logrus with zerolog

### LINKED ISSUE #587: Replacing Logrus with zerolog
Currently, [Logrus](https://github.com/sirupsen/logrus) is being used for logging. When certain levels like Trace are off, program still evaluates these log messages, leading to potential performance issues. For example:

```go
logger.Trace("Processing took: ", complexFunction())
```

In this case, `complexFunction()` is evaluated even if Trace level logging is off, adding unnecessary overhead.
An alternative approach is to use the [zerolog](https://github.com/rs/zerolog) library which provides more control over logging overhead:

```go
log.Trace().Msg("this message may not be logged depending on level", complexFunction())
```

In this case, `complexFunction()` would only be evaluated when the trace log level is enabled, which could improve performance by reducing unnecessary computations and log object creation.

For a performance comparison between zerolog and logrus, check out this [benchmark](http://hackemist.com/logbench/).

### PR DESCRIPTION
**Pull request by kehiy**
## Description
I replaced Logrus with the Zerolog package.

## Related issue(s)

If this Pull Request is related to an issue, mention it here.
- Fixes #587

### CODE REVIEW

**Comment by kehiy** [comment]
@b00f can you review this?

File: go.mod [hunk 1]
```diff
@@ -30,6 +29,8 @@ require (
 	google.golang.org/protobuf v1.30.0
 )
 
+require github.com/mattn/go-colorable v0.1.12 // indirect
```
Comment (b00f) [inline go.mod:32]: Why two requires? I think you can add it in the below lines

File: util/logger/logger.go [hunk 2]
```diff
@@ -130,97 +123,112 @@ func keyvalsToFields(keyvals ...interface{}) logrus.Fields {
...
 }
 
-func (l *Logger) log(level logrus.Level, msg string, keyvals ...interface{}) {
-	if l.logger.IsLevelEnabled(level) {
-		l.withDefaultFields().WithFields(keyvalsToFields(keyvals...)).Log(level, msg)
-	}
-}
-
-func (l *Logger) With(keyvals ...interface{}) *logrus.Entry {
+func (l *Logger) With(keyvals ...interface{}) *zerolog.Event {
```
Comment (b00f) [inline util/logger/logger.go:157]: Do we really need this function? If not, can we delete it?

File: util/logger/logger.go [hunk 3]
```diff
@@ -130,97 +123,112 @@ func keyvalsToFields(keyvals ...interface{}) logrus.Fields {
...
 	fields := keyvalsToFields(keyvals...)
-	return l.logger.WithFields(fields)
+	return l.logger.Log().Fields(fields)
 }
 
 func (l *Logger) Trace(msg string, keyvals ...interface{}) {
-	l.log(logrus.TraceLevel, msg, keyvals...)
+	l.Log(zerolog.TraceLevel, msg, keyvals...)
```
Comment (b00f) [inline util/logger/logger.go:163]: I am wondering is it possible to have something like:
```
l.logger.Trace().Msgf(msg, eyvals...)
```

We get two points:
1- No need to convert keyvals to fields
2- Get rid of the switch-case

**Comment by b00f** [comment]
@kehiy  Some tests are failed 😔.

File: util/logger/logger.go (hunk 3 above)
Comment (kehiy) [reply util/logger/logger.go:163]: what should I do for this?

**Comment by kehiy** [comment]
> @kehiy Some tests are failed pensive.

the logger tests are fine!
but it seems mu changes made some side effects, where? i'm trying to find it.

File: go.mod (hunk 1 above)
Comment (kehiy) [reply go.mod:32]: I used `go mod tidy` 
I will add it to the first one.

File: util/logger/logger.go (hunk 2 above)
Comment (b00f) [reply util/logger/logger.go:157]: If it's only for testing, we can remove it. We haven't needed it yet, and probably won't in the future.

Comment (kehiy) [reply util/logger/logger.go:157]: ok, I will remove it

**Pull request closed without merging**